src/Promitor.Core.Scraping/Configuration/Serialization/Deserializers/GenericAzureMetricDeserializer.cs

```diff
@@ -0,0 +1,43 @@
+﻿using GuardNet;
+using Microsoft.Extensions.Logging;
+using Promitor.Core.Scraping.Configuration.Model.Metrics;
+using Promitor.Core.Scraping.Configuration.Model.Metrics.ResourceTypes;
+using YamlDotNet.RepresentationModel;
+
+namespace Promitor.Core.Scraping.Configuration.Serialization.Deserializers
+{
+    internal class GenericAzureMetricDeserializer
+    {
+        protected ILogger Logger { get; private set; }
+
+        internal GenericAzureMetricDeserializer WithLogger(ILogger logger)
+        {
+            Logger = logger;
+            return this;
+        }
+
+        internal virtual MetricDefinition Deserialize(YamlMappingNode metricNode) => DeserializeInternal<GenericAzureMetricDefinition>(metricNode);
+
+        protected virtual TMetricDefinition DeserializeInternal<TMetricDefinition>(YamlMappingNode metricNode)
```

Comment:
Not really a fan of methods call `DeserializeInternal`, certainly given it's protected and not internal.

How about `DeserializeMetricDefinition`?